doc/governance/proposals.rst

```diff
@@ -46,76 +46,130 @@ For transparency and auditability, all governance operations (including votes) a
 Creating a Proposal
 -------------------
 
-For custom proposals with multiple actions and precise conditional requirements you will need to write the proposal script by hand.
-For simple proposals there is a helper script in the CCF Python package - ``proposal_generator.py``.
-This can be used to create proposals for common operations like adding members and users, without writing any JSON.
-It also produces sample vote scripts, which validate that the executed proposed actions exactly match what is expected.
-These sample proposals and votes can be used as a syntax and API reference for producing more complex custom proposals.
-
-Assuming the CCF Python package has been installed in the current Python environment, the proposal generator can be invoked directly as ``ccf.proposal_generator``. With no further argument it will print help text, including the list of possible actions as subcommands:
+Proposals are JSON objects specifying a list of actions and associated arguments. The name of the action should match one of the actions available in your service (in ``actions.js``), and the expected types of the arguments are dictated by the validate and apply functions which handle that action.
+These proposal objects can be constructed by an tool or language which can produce JSON. The CCF pip package contains a sample of a tool to automate proposal creation, written in bash and wrapping the CLI tool ``jq``, named ``build_proposal.sh``:
 
 .. code-block:: bash
 
-    usage: proposal_generator.py [-h] [-po PROPOSAL_OUTPUT_FILE] [-vo VOTE_OUTPUT_FILE] [-pp] [-i] [-v]
-                                {add_node_code,remove_ca_cert_bundle,remove_js_app,remove_jwt_issuer,remove_member,remove_node,remove_node_code,remove_user,set_ca_cert_bundle,set_constitution,set_js_app,set_jwt_issuer,set_jwt_public_signing_keys,set_member,set_member_data,set_recovery_threshold,set_user,set_user_data,transition_node_to_trusted,transition_service_to_open,trigger_ledger_rekey,trigger_recovery_shares_refresh}
+    $ build_proposal.sh --help
 
-Additional detail is available from the ``--help`` option. You can also find the script in a checkout of CCF:
+    Usage:
+      build_proposal.sh [--help | --action ACTION_NAME [[FLAGS] ARG_NAME ARG_VALUE...]...]
 
-.. code-block:: bash
+    This tool is a wrapper around jq, to simplify creation of CCF governance
+    proposals.
+    Specify a list of actions and associated args. A single flag per argument can be
+    used to indicate how the value should be parsed:
+      -s String (default)
+      -j JSON (including raw numbers)
+      -b Boolean (including case-insensitive parsing)
```

Comment:
If the only legal values are "true", "TRUE", "false" and "FALSE", we might as well list them here, and avoid any possible assumptions about 0, 1, yes, no etc (I know, and yet YAML does it).

Reply:
It also handles `True` and `False` (so that I could be lazy when calling this from Python), and by accident `tRuE` and all such dumb variations.

Reply:
Actually saying that, I don't think that's needed anymore, since I dump most interesting things through `json.dumps` anyway. I'll see if I can just remove this `-b` type entirely

Reply:
Ok, that'd be even better.